File: BOJ/FastCampas_Algorithm_lec/BOJ_1966/프린터큐1966.cpp
#include<iostream>
#include<queue>
using namespace std;
int max(int a, int b) {
	return a > b ? a : b;
}
int main() {
	int tc;
	cin >> tc;

	while (tc--) {
		int cnt = 0, maxi = 0;
		queue<pair<int, int> > q;
		priority_queue<int> pq;
		int n, m, pri;

		cin >> n >> m;
		for (int i = 0; i < n; i++) {
			cin >> pri;
			pq.push(pri);
			q.push(make_pair(pri, i));
		}
		while (1) {
			maxi = pq.top();
			if (q.front().first == maxi) {
				cnt += 1;
				if (q.front().second == m) {
					cout << cnt << "\n";
					break;
				}
				else {
					q.pop();
					pq.pop();
				}
			}
			else {
				int a = q.front().first;
				int b = q.front().second;
				q.pop();
				q.push(make_pair(a, b));
			}
		}
	}
	return 0;
}

File: BOJ/FastCampas_Algorithm_lec/BOJ_1966/프린터큐1966.py
tc = int(input())

for _ in range(tc):
    n, m = list(map(int, input().split(' ')))
    queue = list(map(int, input().split(' ')))
    queue = [(i ,idx) for idx, i in enumerate(queue)]

    cnt = 0
    while True:
        if queue[0][0] == max(queue, key=lambda x: x[0])[0]:
            cnt += 1
            if queue[0][1] == m:
                print(cnt)
                break
            else:
                queue.pop(0)
        else:
            queue.append(queue.pop(0))